File: MergeIntervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        if(intervals.length <= 1){
            return intervals;
        }
        
        // Sorting 2D Array Using Its First Element In Every Row
        Arrays.sort(intervals, (arr1, arr2) -> Integer.compare(arr1[0], arr2[0]));
        
        //Using ArrayList To Store Final Output Because Size Of Result Is Unknown.
        List<int[]> result = new ArrayList<>();
        
        // Adding First Row In The Result ArrayList
        int[] currentElementArray = intervals[0];
        result.add(currentElementArray);
        
        // Traversing Through The Array And Comparing Current End To Next Start
        for(int[] interval : intervals){

            int currentStart = currentElementArray[0];
            int currentEnd = currentElementArray[1];
            int nextStart = interval[0];
            int nextEnd = interval[1];
            
            if(currentEnd >= nextStart){
                currentElementArray[1] = Math.max(currentEnd, nextEnd);
            }else{
                currentElementArray = interval;
                result.add(currentElementArray);
            }
            
        }
        
        // Returning A 2D Array As The Result (Return Type Being A 2D Array)
        return result.toArray(new int[result.size()][]);
        
    }
}

File: MergeIntervals.js
/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */



var merge = function(intervals) {
    // Base Condition
    if(!intervals.length) return intervals;
    // Sort Based On The First Start Interval Of The Array
    intervals.sort((a,b) => a[0] - b[0]);

    let prev = intervals[0];
    let result = [prev];

    const mergeIntervals = (currentElement) => {
        if(currentElement[0] <= prev[1]){
            prev[1] = Math.max(prev[1], currentElement[1]);
        }
        else {
            result.push(currentElement);
            prev = currentElement;
        }
    }


    // Creating the new intervals
    intervals.forEach(mergeIntervals);
    return result;
};
